app/extension/bitcoinInterface.js

```diff
@@ -0,0 +1,62 @@
+const bitcoin = require('../bitcoin/bitcoin.js');
+const persistence = require('../bitcoin/persistence.js');
+
+const address = new bitcoin.BitcoinAddress(persistence);
+let transfer = new bitcoin.BitcoinTransfer(address);
+
+
+function addDonation() {
+    if (!transfer.sufficientInput()) {  //If current input isn't enough, attempt to add more before proceeding
+        const unusedInputs = address.requestTransactionList()   //Get list of all unspent transactions associated with this address
+            .map(output => ({tx_hash: output.tx_hash, tx_index: output.tx_index, value: output.value}))
+            .filter((output) => !transfer.inputs.some((input) => {  //Filter based on what ones are already in use for building the current transaction
```

Comment:
"based on what ones are already in use" -> "based on what inputs are already in use"?